services/autoscaling/tests/unit/test_modules_auto_scaling_computational.py

```diff
@@ -1029,3 +1030,159 @@ async def test_cluster_scaling_up_more_than_allowed_with_multiple_types_max_star
+    )
+    assert dask_future
+    # this should trigger a scaling up as we have no nodes
+    await auto_scale_cluster(
+        app=initialized_app, auto_scaling_mode=ComputationalAutoscaling()
+    )
+
+    # check the instance was started and we have exactly 1
+    instances = await _assert_ec2_instances(
+        ec2_client,
+        num_reservations=1,
+        num_instances=1,
+        instance_type=expected_ec2_type,
+        instance_state="running",
+    )
+
+    # as the new node is already running, but is not yet connected, hence not tagged and drained
+    mock_find_node_with_name_returns_none.assert_not_called()
+    mock_docker_tag_node.assert_not_called()
+    # check rabbit messages were sent
+    _assert_rabbit_autoscaling_message_sent(
+        mock_rabbitmq_post_message,
+        app_settings,
+        initialized_app,
+        dask_spec_local_cluster.scheduler_address,
+        instances_running=0,
+        instances_pending=1,
+    )
+    mock_rabbitmq_post_message.reset_mock()
+
+    assert instances
+    assert "LaunchTime" in instances[0]
+    original_instance_launch_time: datetime.datetime = deepcopy(
+        instances[0]["LaunchTime"]
+    )
+    await asyncio.sleep(1)  # NOTE: we wait here since AWS does not keep microseconds
+    now = arrow.utcnow().datetime
+
+    assert now > original_instance_launch_time
+    assert now < (
+        original_instance_launch_time
+        + app_settings.AUTOSCALING_EC2_INSTANCES.EC2_INSTANCES_MAX_START_TIME
+    )
+
+    # 2. running again several times the autoscaler, the node does not join
+    for i in range(7):
+        await auto_scale_cluster(
+            app=initialized_app, auto_scaling_mode=ComputationalAutoscaling()
+        )
+        # there should be no scaling up, since there is already a pending instance
+        instances = await _assert_ec2_instances(
+            ec2_client,
+            num_reservations=1,
+            num_instances=1,
+            instance_type=expected_ec2_type,
+            instance_state="running",
+        )
+        assert mock_find_node_with_name_returns_none.call_count == i + 1
+        mock_docker_tag_node.assert_not_called()
+        _assert_rabbit_autoscaling_message_sent(
+            mock_rabbitmq_post_message,
+            app_settings,
+            initialized_app,
+            dask_spec_local_cluster.scheduler_address,
+            instances_running=0,
+            instances_pending=1,
+        )
+        mock_rabbitmq_post_message.reset_mock()
+        assert instances
+        assert "LaunchTime" in instances[0]
+        assert instances[0]["LaunchTime"] == original_instance_launch_time
+
+    # 3. wait for the instance max start time and try again, shall terminate the instance
+    now = arrow.utcnow().datetime
+    sleep_time = (
+        original_instance_launch_time
+        + app_settings.AUTOSCALING_EC2_INSTANCES.EC2_INSTANCES_MAX_START_TIME
+        - now
+    ).total_seconds() + 1
+    print(
```

Comment:
question for my learning: in pytests one prints instead of using loggers? is there a reason?

Reply:
well when you run ```pytest -s``` it shows prints, otherwise you run ```pytest --log-cli-level=INFO``` and then you see the logs. most of the time in simcore we used print in tests. not really sure why.